bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -486,9 +492,10 @@ public ScanHandler readFlatRows(ReadRowsRequest request, StreamObserver<FlatRow>
   }
 
   private RetryingReadRowsOperation createReadRowsRetryListener(ReadRowsRequest request,
-      StreamObserver<FlatRow> observer) {
+      StreamObserver<FlatRow> rowObserver, StreamObserver<ReadRowsResponse> responseObserver) {
```

Comment:
This looks really weird, it seems like we are breaking encapsulation here

Reply:
Do you mean that that having a `StreamObserver<ReadRowsResponse>` is a problem in general, or that we set it in the constructor?

Reply:
Sorry, I didn't do a very good job communicating.

This PR promotes the raw response observer to the same level as the row one. The 2 observers listen for events from 2 different layers, so there some mixing of concerns here. Also, there is nothing to tell the reader why there are 2 observers and what each one is used for.  It's also a bit arbitrary that one of them nullable...there is nothing telling the reader that the response one is optional. Before the PR there was some separation: the primary row observer was a toplevel parameter, while the optional response observer was relegated to an optional setter. 

At the very least, adding some docs and maybe better method name would help. 

I'm not sure if this would improve the code,  but maybe it would make sense to convert the response observer into a less generic signal like a `onRowBufferEmpty` (not sure about the name) method on the row observer. Another option is that the end result of receiving a notification of a response is to insert a marker into the buffer alongside the logical rows. Maybe have the operation send the marker itself?